src/js/components/Articles/modules/EpubView/EpubView.js

```diff
@@ -0,0 +1,136 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import Epub from "epubjs/lib/index";
+import defaultStyles from "./style";
+
+global.ePub = Epub; // Fix for v3 branch of epub.js -> needs ePub to by a global var
+
+class EpubView extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isLoaded: false,
+      toc: []
+    };
+    this.viewerRef = React.createRef();
+    this.location = props.location;
+    this.book = this.rendition = this.prevPage = this.nextPage = null;
+  }
+
+  componentDidMount() {
+    const { url, tocChanged } = this.props;
+    // use empty options to avoid ArrayBuffer urls being treated as options in epub.js
+    const epubOptions = {};
+    this.book = new Epub(url, epubOptions);
+    this.book.loaded.navigation.then(({ toc }) => {
+      this.setState(
+        {
+          isLoaded: true,
+          toc: toc
+        },
+        () => {
+          tocChanged && tocChanged(toc);
+          this.initReader();
+        }
+      );
+    });
+    document.addEventListener("keydown", this.handleKeyPress, false);
+  }
+
+  componentWillUnmount() {
```

Comment:
Good practice used here.
Keep this lifecycle method in mind to clean up your React component before they are unmounted and destroyed.
Beautiful!